### PR TITLE
uboot: Use the uboot support provided by meta-balena for cmdline

### PR DESCRIPTION
**Pull request by agherzan**
This removes the use of OS_KERNEL_CMDLINE directly by switching to the
uboot support provided by `os_cmdline` variable set by `set_os_cmdline`.

Signed-off-by: Andrei Gherzan <andrei@balena.io>

_**Please don't merge until https://github.com/balena-os/meta-balena/pull/1506 gets merged and an update of the meta-balena version which includes these changes is added to this PR.**_

### CODE REVIEW

File: layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/0004-Integrate-with-Balena-environment-configuration.patch [hunk 1]
```diff
@@ -5,6 +5,7 @@ Subject: [PATCH] Integrate with Balena environment configuration
 
 Upstream-Status: Inappropriate [configuration]
 Signed-off-by: Florin Sarbu <florin@balena.io>
+Signed-off-by: Andrei Gherzan <andrei@balena.io>
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/0004-Integrate-with-Balena-environment-configuration.patch]: Needs `Changelog-entry`

Comment (agherzan) [reply layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/0004-Integrate-with-Balena-environment-configuration.patch]: Added

File: layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/0005-env_default.h-Add-support-for-OS_KERNEL_CMDLINE.patch [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
- 	ENV_CALLBACK_VAR "=" CONFIG_ENV_CALLBACK_LIST_DEFAULT "\0"
+diff --git a/include/environment/ti/mmc.h b/include/environment/ti/mmc.h
+index 1bf2668a16..ea64a6c0fe 100644
+--- a/include/environment/ti/mmc.h
++++ b/include/environment/ti/mmc.h
+@@ -19,6 +19,7 @@
+ 		"root=PARTUUID=${uuid} ro " \
+ 		"rootfstype=${mmcrootfstype} " \
+ 		"${uboot_detected_capes} " \
++		"${os_cmdline} " \
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/0005-env_default.h-Add-support-for-OS_KERNEL_CMDLINE.patch]: I think our `${os_cmdline}` should be after `${cmdline}` here

Comment (agherzan) [reply layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/0005-env_default.h-Add-support-for-OS_KERNEL_CMDLINE.patch]: What is the reason for that? the cmdline is empty anyway and having the same argument multiple times behaves differently based on the module.

Comment (floion) [reply layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/0005-env_default.h-Add-support-for-OS_KERNEL_CMDLINE.patch]: If it's empty then it's no problem. I was under the impression it was defined with something. Did you check in the running u-boot shell that it actually ends up empty?

Comment (agherzan) [reply layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot/0005-env_default.h-Add-support-for-OS_KERNEL_CMDLINE.patch]: Yes. It is always empty.

**Comment by floion** [comment]
@balena-ci retest

**Pull request merged**